src/index.ts

```diff
@@ -874,7 +885,17 @@ class ThreadPool {
   }
 
   _maybeDrain () {
-    if (this.taskQueue.size === 0 && this.skipQueue.length === 0) {
+    /**
+     * Our goal is to make it possible for user space to use the pool
+     * in a way where always waiting === 0, since we want to avoid creating tasks that can't execute
+     * immediately in order to provide back pressure to the task source.
+     */
+    const maxCapacity = this.options.maxThreads * this.options.concurrentTasksPerWorker;
+    const currentUsage = this.workers.getCurrentUsage();
+
+    if (maxCapacity === currentUsage) {
+      // TODO: needs drain goes here
+    } else if (maxCapacity > currentUsage) {
```

Comment:
```suggestion
    /**
     * Our goal is to make it possible for user space to use the pool
     * in a way where always waiting === 0, since we want to avoid creating tasks that can't execute
     * immediately in order to provide back pressure to the task source.
     */
    const maxCapacity = this.options.maxThreads * this.options.concurrentTasksPerWorker;
    const currentUsage = this.workers.getCurrentUsage();

    if (currentUsage  >= maxCapacity && !this._needsDrain) {
      this._needsDrain = true
    } else if (currentUsage < maxCapacity && this._needsDrain) {
```